Add StoreResource with CRUD operations and improve test coverage

marierinden opened the pull request:
### Description of changes

- Introduced `StoreResource`, enabling CRUD operations for stores with robust validation, mapping, and error handling.
- Added unit tests for `StoreResponse`, `StoreCollectionResponse`, and `StoreRequest` to enhance test coverage.
- Removed the unused `organization` field from `StoreResponse` for better alignment with current requirements.

marierinden committed:
Add store resource with CRUD operations

Introduce `StoreResource` providing CRUD operations for stores. This includes request and response DTOs, supporting structured interactions with the store API. The implementation ensures validation, mapping, and error handling across all operations.

marierinden committed:
Add unit tests for Store DTOs and remove unused field

Added unit tests for StoreResponse, StoreCollectionResponse, and StoreRequest to improve test coverage. Removed the unused 'organization' field from StoreResponse as it is no longer needed. These changes ensure alignment between the DTOs and their required functionality.

fredrik-tveraaen-apiera commented on src/DTO/Request/Store/StoreRequest.php:
```diff
@@ -0,0 +1,94 @@
+     */
+    public function __construct(
+        #[RequestField('name')]
+        private string $name,
```
Change to nullable

fredrik-tveraaen-apiera commented on src/DTO/Request/Store/StoreRequest.php:
```diff
@@ -0,0 +1,94 @@
+        #[RequestField('name')]
+        private string $name,
+        #[RequestField('organization')]
+        private string $organization,
```
Remove this it's an documentation bug

fredrik-tveraaen-apiera commented on src/DTO/Request/Store/StoreRequest.php:
```diff
@@ -0,0 +1,94 @@
+        private ?string $description = null,
+        #[RequestField('image')]
+        private ?string $image = null,
+        #[RequestField('properties')]
+        private array $properties = [],
+        #[RequestField('propertyTerms')]
+        private array $propertyTerms = [],
```
Remove this it's an documentation bug

fredrik-tveraaen-apiera commented on src/Resource/StoreResource.php:
```diff
@@ -0,0 +1,167 @@
+ */
+final readonly class StoreResource implements RequestResourceInterface
+{
+    private const string ENDPOINT = '/stores';
```
Add prefix /api/v1/ as stores is not a nested endpoint

marierinden committed:
Refactor StoreRequest to simplify structure and update endpoint

Removed unused fields and methods, including organization and propertyTerms, from StoreRequest. Adjusted default values for certain fields and updated the Store endpoint to '/api/v1/stores'. Corresponding test cases were updated to reflect these changes.

marierinden committed:
Remove 'properties' and 'propertyTerms' from Store DTOs

The 'properties' and 'propertyTerms' fields were removed from StoreRequest and StoreResponse DTOs, along with their associated methods and tests. This simplifies the data structures and eliminates unused fields, aligning them with updated requirements.

fredrik-tveraaen-apiera committed:
Update StoreResponse and tests to reflect Store as ldType

Changed `ldType` from Product to Store in StoreResponse and its related test cases to align with the correct representation. Made `name` non-nullable and updated the getter accordingly, ensuring consistency with the updated class structure.

fredrik-tveraaen-apiera committed:
Add store handling to SDK

Added support for `Store` type responses and resources in the SDK. This includes mapping `StoreResponse` and `StoreCollectionResponse` in `LdType` and adding a `store()` method in `ApieraSdk` to retrieve `StoreResource` instances.

fredrik-tveraaen-apiera committed:
Refactor StoreResponse fields to use ResponseField attributes.

Replaced RequestField attributes with ResponseField for consistency and accuracy in mapping response data. Removed unused RequestField import to clean up the code. This streamlines the DTO for better alignment with API response handling.

fredrik-tveraaen-apiera committed:
Add integration tests for Store resource operations

Introduce integration tests for Create, Get, Update, and Delete operations for the Store resource. Each test validates functionality, mock responses, and resource-specific assertions for consistent API behavior.

fredrik-tveraaen-apiera committed:
Merge origin release/0.3.0

fredrik-tveraaen-apiera reviewed (approved):
:100:
